### PR TITLE
server: defend panic for ill-packet with null connection attribute

### PR DESCRIPTION
**Pull request by tiancaiamao**
if client say it has connection attribute capability, but don't provide the data, TiDB can ignore it and should not panic.

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -235,6 +235,10 @@ func handshakeResponseFromData(packet *handshakeResponse41, data []byte) error {
 	}
 
 	if capability&mysql.ClientConnectAtts > 0 {
+		if len(data[pos:]) <= 0 {
```
Comment (coocood) [inline server/conn.go]: len() can not be less than 0

**Comment by zimulala** [comment]
Add some tests.

**Comment by tiancaiamao** [comment]
construct the input packet is not easy, previous test data take from mysql document example,
but there is no such wrong packet example.
 @zimulala

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**